test/extensions/filters/http/admission_control/config_test.cc

```diff
@@ -75,7 +75,34 @@ sampling_window: 1337s
   NiceMock<Server::Configuration::MockFactoryContext> factory_context;
   EXPECT_THROW_WITH_MESSAGE(admission_control_filter_factory.createFilterFactoryFromProtoTyped(
                                 proto, "whatever", factory_context),
-                            EnvoyException, "Success Rate Threshold cannot be zero percent");
+                            EnvoyException, "Success Rate Threshold cannot be less than one.");
+}
+
+TEST_F(AdmissionControlConfigTest, SmallSuccessRateThreshold) {
+  AdmissionControlFilterFactory admission_control_filter_factory;
+  const std::string yaml = R"EOF(
+enabled:
+  default_value: false
+  runtime_key: "foo.enabled"
+sampling_window: 1337s
+sr_threshold:
+  default_value:
+    value: 1.22e-22
+  runtime_key: "foo.sr_threshold"
+aggression:
+  default_value: 4.2
+  runtime_key: "foo.aggression"
+success_criteria:
+  http_criteria:
+  grpc_criteria:
+)EOF";
+
+  AdmissionControlProto proto;
+  TestUtility::loadFromYamlAndValidate(yaml, proto);
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
+  EXPECT_THROW_WITH_MESSAGE(admission_control_filter_factory.createFilterFactoryFromProtoTyped(
+                                proto, "whatever", factory_context),
+                            EnvoyException, "Success Rate Threshold cannot be less than one.");
```

Comment:
ah I see here that we are just worried about it being a small positive float. OK, then I'd write "less than 1.0".

Reply:
Yep! A small positive float can eventually round down to a division by zero. 
https://github.com/envoyproxy/envoy/blob/846f27c58aac9ea88b0a85504791674fd9934d82/source/extensions/filters/http/admission_control/admission_control.cc#L145

The alternative is to round up in this division but I don't want to change behavior for reasonable values (above 1)
 https://github.com/envoyproxy/envoy/blob/846f27c58aac9ea88b0a85504791674fd9934d82/source/extensions/filters/http/admission_control/admission_control.cc#L56

Reply:
Just looping @tonya11en @mattklein123 
The bounds for threshold described here say [0, 1]. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/admission_control_filter
Should it actually be (0, 1] to avoid undefined behavior -- and in practice should it be [0.01, 1.0] (which is what this fix provides, blocking success rate threshold below 1.0%?)

Reply:
Yeah I think this is reasonable. :+1: